source/slang/slang-preprocessor.cpp

```diff
@@ -763,7 +1136,13 @@ static PreprocessorMacro* _parseMacroArg(Preprocessor* preprocessor)
     int nestingDepth = 0;
     for(;;)
     {
-        switch(PeekRawTokenType(preprocessor))
+        arg.endTokenIndex = macroInvocation->m_argTokens.getCount();
+
+        m_inputStreams.skipAllWhitespace();
```

Comment:
Yep, I agree that it feels a little unclean. This was the best compromise across all the alternatives I mocked up during development of the PR.

In my mind the right way to clean this up (while retaining a model where the lexer returns both semantically-relevant tokens and trivia) is to flip the mental model.

Right now the code things primarily in terms of a "read/peek a token" operation that returns either a semantically-relevant token or trivia, and then has to special case the "skip whitespace, then read/peek a token" operation that only returns the semantically-relevant tokens.

A lot of the problems go away with the fundamental operation is "read/peek a semantically/relevant token" and then there is a special case operation that is either "read/peek a token or trivia" *or* "get me all the trivia related to this semantically-relevant token."

I really want to get to that cleanup, since I suspect it could make a lot of things nicer.